src/libcore/macros.rs

```diff
@@ -108,6 +108,44 @@ macro_rules! assert_eq {
     });
 }
 
+/// Asserts that two expressions are not equal to each other.
+///
+/// On panic, this macro will print the values of the expressions with their
+/// debug representations.
+///
+/// # Examples
+///
+/// ```
+/// let a = 3;
+/// let b = 2;
+/// assert_ne!(a, b);
+/// ```
+#[macro_export]
+///#[stable(feature = "????", since = "????")]
+macro_rules! assert_eq {
+    ($left:expr , $right:expr) => ({
+        match (&$left, &$right) {
+            (left_val, right_val) => {
+                if *left_val == *right_val {
+                    panic!("assertion failed: `(left != right)` \
```

Comment:
Here though the assertion that's failing is that `(left == right)`, not `(left != right)`, right?

Otherwise, if I write `assert_ne!(1, 1)` it'll print:

```
assertion failed: (1 != 1)
```